editor/find_in_files.cpp

```diff
@@ -381,18 +392,50 @@ FindInFilesDialog::FindInFilesDialog() {
 
 	Button *cancel_button = get_ok();
 	cancel_button->set_text(TTR("Cancel"));
+
+	_mode = SEARCH_MODE;
 }
 
 void FindInFilesDialog::set_search_text(String text) {
 	_search_text_line_edit->set_text(text);
 	_on_search_text_modified(text);
 }
 
+void FindInFilesDialog::set_replace_text(String text) {
+	_replace_text_line_edit->set_text(text);
+}
+
+void FindInFilesDialog::set_find_in_files_mode(FindInFilesMode p_mode) {
+
+	if (_mode == p_mode)
+		return;
+
+	_mode = p_mode;
+
+	if (p_mode == SEARCH_MODE) {
+		set_title(TTR("Find in Files"));
+		_replace_label->hide();
+		_replace_text_line_edit->hide();
+	} else if (p_mode == REPLACE_MODE) {
+		set_title(TTR("Replace in Files"));
+		_replace_label->show();
+		_replace_text_line_edit->show();
+	}
+
+	//	After hiding some child controls, let's recalculate proper Dialog size
+	set_size(Size2(get_size().x, 0));
+}
+
 String FindInFilesDialog::get_search_text() const {
 	String text = _search_text_line_edit->get_text();
 	return text.strip_edges();
 }
 
+String FindInFilesDialog::get_replace_text() const {
+	String text = _replace_text_line_edit->get_text();
+	return text.strip_edges();
```

Comment:
Should we be changing the replace text?

Reply:
Well, not programaticaly (at least - there is no need currently for that).
Method was added to allow `void ScriptEditor::_start_find_in_files(bool with_replace)` to get nicely that text from the Control.

Reply:
I meant the `strip_edges` on the text.

Reply:
Oh, you're right - I was aligning that with `String FindInFilesDialog::get_search_text() const` - for which removing spaces may have sense.
If that's OK with you, I can alter that slightly and remove that modification, to preserve literaly-typed text as a replacement.

Reply:
Sounds good to me :)

Reply:
Done :)